Documentation style changes for the "Installation" section (#6015)
nginxinc/kubernetes-ingress: docs/content/installation/run-multiple-ingress-controllers.md

docs: DOCS-606
doctypes:
- ''
title: Run multiple NGINX Ingress Controllers
toc: true
weight: 400
---

This document describes how to run multiple F5 NGINX Ingress Controller instances.

It explains the following topics:

- Ingress class concept.
- How to run NGINX Ingress Controller in the same cluster with another Ingress Controller and prevent conflicts between them
- How to run multiple NGINX Ingress Controllers.

{{< note >}} This document refers to [Ingress]({{< relref "/configuration/ingress-resources/basic-configuration.md" >}}), [VirtualServer]({{< relref "/configuration/virtualserver-and-virtualserverroute-resources.md#virtualserver-specification" >}}), [VirtualServerRoute]({{< relref "/configuration/virtualserver-and-virtualserverroute-resources.md#virtualserverroute-specification" >}}), and [TransportServer]({{< relref "/configuration/transportserver-resource.md" >}}) resources as "configuration resources".{{< /note >}}

---

## Ingress class

> Due to the design of the Ingress class, multiple Ingress Controllers can coexist in one cluster. The Ingress class has the following characteristics:

Reviewer: ```suggestion
The [IngressClass](https://kubernetes.io/docs/concepts/services-networking/ingress/#ingress-class) resource allows for multiple Ingress Controller to operate in the same cluster. It also allow developers to select which Ingress Controller implementation to use for their Ingress resource.
The IngressClass has the following characteristics:
```

{{< note >}}- If the class of an Ingress resource is not set, Kubernetes will set it to the class of the default Ingress Controller. To make the Ingress Controller the default one, the `ingressclass.kubernetes.io/is-default-class` property must be set on the IngressClass resource. To learn more, see Step 3 *Create an IngressClass resource* of the [Create Common Resources]({{< relref "installation/installing-nic/installation-with-manifests.md#create-common-resources" >}}) section.
- For VirtualServer, VirtualServerRoute, Policy and TransportServer resources, NGINX Ingress Controller will always handle resources with an empty class.{{< /note >}}

---

## Run NGINX Ingress Controller and another Ingress Controller

It is possible to run NGINX Ingress Controller and an Ingress Controller for another load balancer in the same cluster. This is often the case if you create your cluster through a cloud provider's managed Kubernetes service that by default might include the Ingress Controller for the HTTP load balancer of the cloud provider, and you want to use NGINX Ingress Controller.

> To make sure that NGINX Ingress Controller handles specific configuration resources, update those resources with the class set to `nginx` (or the Ingress class value that you configured in NGINX Ingress Controller).

Reviewer: ```suggestion
To make sure that NGINX Ingress Controller handles specific configuration resources, update those resources with the class set to the value that is configured in NGINX Ingress Controller. By default, this is `nginx`
```
